server/src/main/java/org/elasticsearch/cluster/routing/UnassignedInfo.java

```diff
@@ -241,7 +242,7 @@ public UnassignedInfo(Reason reason, @Nullable String message, @Nullable Excepti
         this.failure = failure;
         this.failedAllocations = failedAllocations;
         this.lastAllocationStatus = Objects.requireNonNull(lastAllocationStatus);
-        this.failedNodeIds = Collections.unmodifiableSet(failedNodeIds);
+        this.failedNodeIds = Collections.unmodifiableSet(new HashSet<>(failedNodeIds));
```

Comment:
I think we can just use `Set.copyOf(` here to be shorter and have a singleton for the empty case.

Reply:
@original-brownbear   Thank you very much for your suggestion. I thought about this at that time. Jdk12, 13, 14 and 15 are short-term versions. Therefore, enterprises and applications generally use jdk8 or jdk11. I saw that in elasticsearch, the minimumRuntimeVersion file is set to 11, and Set.copyof() fails to compile in jdk11.

Reply:
Not sure what you mean. `java.util.Set#copyOf` is available from JDK 10 and `public`. In `master` we do not support any JDK version older than `10` so it's not a problem there and for the `7.x` backport we do have a wrapper in place that we can use. We use that utility all over the codebase and I don't see a reason why it wouldn't work here?

Reply:
@original-brownbear Yes, you're right. Thank you very much. You're great. I've solved it.